core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java

```diff
@@ -244,7 +244,11 @@ public long executeUpdate(SqlExecutionContext sqlExecutionContext, UpdateOperati
             tableWriter.rollbackUpdate();
             throw th;
         } finally {
-            op.closeWriter();
+            if (sqlExecutionContext.isWalApplication()) {
```

Comment:
I think here it should call `closeWriter()` unconditionally as before.
To fix resource leak, `ApplyWal2TableJob` should close the UpdateOperation.

Reply:
best would be to remove async mode completely :)
will have a look at closing it from apply job.

Reply:
changed it